lib/ApiRequestor.php

```diff
@@ -278,6 +278,10 @@ private static function _specificV2APIError($rbody, $rcode, $rheaders, $resp, $e
 
             // The end of the section generated from our OpenAPI spec
             default:
+                if ('invalid_fields' === $code) {
+                    $param = isset($errorData['param']) ? $errorData['param'] : null;
+                    return Exception\InvalidRequestException::factory($msg, $rcode, $rbody, $resp, $rheaders, $code, $param);
+                }
```

Comment:
Wouldn't it better if we moved this to `_specificV1APIError` under 400 status code?

Reply:
That would make it consistent with all languages except node.

Reply:
That would be a pretty breaking change for a lot of users - I think this minimizes the scope a lot more. I don't know that it's worth changing how exceptions work for all v1 users here for not much gain (imo). Curious for your opinion, though.

Reply:
I'd assume that they would be catching a more general `ApiErrorException` anyways. It would be nicer for PHP to also have more accurate exception type and a more consistent error responses with the rest of the languages. 

This is a major release with probably a migration guide too so I'd say most people wouldn't mind putting in the effort to rename and make the change.

Reply:
I'm going to ask Remi real quick, because my feel is that this is pretty breaking.